### ISSUE TITLE
Error: parsing segment "resourceGroups": expected the segment "resourcegroups" to be "resourceGroups"

### ISSUE THREAD

**Issue by mohammadkandy** [hashicorp/terraform-provider-azurerm#16518]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.7

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

azurerm_user_assigned_identity

### Terraform Configuration Files

```hcl
resource "azurerm_user_assigned_identity" "user_identity" {
  name                = module.naming.name
  resource_group_name = var.resource_group_name
  location            = var.location
}

resource "azurerm_application_gateway" "app_gateway" {

...

  identity {
    identity_ids = [azurerm_user_assigned_identity.user_identity.id]
    type         = var.identity_type
  }

...

}
```

### Debug Output/Panic Output

```shell
╷
│ Error: parsing "/subscriptions/*/resourcegroups/*/providers/Microsoft.ManagedIdentity/userAssignedIdentities/d-ext-agw01": 
│ parsing segment "resourceGroups": expected the segment "resourcegroups" to be "resourceGroups"
│
╵
Releasing state lock. This may take a few moments...
ERRO[0022] 1 error occurred:
        * exit status 1
```

### Expected Behaviour

Managed Identity ID's are parsed correctly: "resourceGroups"

### Actual Behaviour

Managed Identity ID's are not parsed correctly: "resourcegroups"

### Steps to Reproduce

Attempt to deploy a user assigned identity with app gateway.

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/14456
https://github.com/hashicorp/terraform-provider-azurerm/blob/v2.88.1/CHANGELOG.md#2881-december-03-2021

**Comment by ms-henglu** [comment]
Hi @mohammadkandy ,

Thank you for taking time to report this issue.

It seems in the latest azurerm release, `azurerm_user_assigned_identity`'s id is in a correct format, would you please try to recreate the user assigned identity with https://github.com/hashicorp/terraform-provider-azurerm/releases/tag/v3.3.0?

**Comment by mohammadkandy** [comment]
@ms-henglu Thank you for this, can confirm the issue is now resolved after upgrading to `azurerm 3.3.0`.
